odoo/orm/fields_relational.py

```diff
@@ -100,8 +104,12 @@ def validated(domain):
     _description_relation = property(attrgetter('comodel_name'))
     _description_context = property(attrgetter('context'))
 
-    def _description_domain(self, env):
-        domain = self.domain(env[self.model_name]) if callable(self.domain) else self.domain  # pylint: disable=not-callable
+    def _description_domain(self, env) -> str | list:
+        domain = self.domain
+        if callable(domain):
+            domain = domain(env[self.model_name])
+        if not isinstance(domain, str):
+            domain = list(self.get_comodel_domain(env[self.model_name]))
```

Comment:
That's weird conditions, it means that in case of callable domain, you call the `domain` method twice. IMO, you shouldn't use `get_comodel_domain` here.

Reply:
Yes it is called twice because we must call `get_comodel_domain` when the domain is not a string (including if the function returns a string). The `get_comodel_domain` may have overwrites to add additional logic (ie: o2m reference).

Reply:
Code updated